docs/blogs/EnhancedDevPreviewGatewayAPI/GettingCreative.md

```diff
@@ -0,0 +1,294 @@
+a way to match traffic on any of these attributes.
+
+The `headers` match rule can be used to match on custom headers or on standard HTTP headers. For example, a rule could
+match on the `accept-encoding` header in order to use a different backend for clients requesting HTML and clients requesting
+json. Similarly, `queryParams` can match on arbitrary URL query parameters.
+
+The following example shows header and URL query parameter matching. First, we set a match rule using `queryParams` that
+if the `app` query parameter is set to `foo`, the request should be forwarded to the `foo-app` application; and second, we
+set a match rule using headers that if the `x-app` header is set to `bar`, the request should be forwarded to the `bar-app`
+application:
+```console
+$ oc apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: http
+  namespace: demo-app
+spec:
+  parentRefs:
+  - name: demo-gateway
+    namespace: openshift-ingress
+  hostnames: ["http.gwapi.${DOMAIN}"]
+  rules:
+  - matches:
+    - queryParams: #[1]
+      - name: app
+        value: foo
+    backendRefs:
+    - name: foo-app
+      port: 8080
+  - matches:
+    - headers: #[2]
+      - name: x-app
+        value: bar 
+    backendRefs:
+    - name: bar-app
+      port: 8080
+EOF
+httproute.gateway.networking.k8s.io/http configured
+```
+[1] `queryParams`: if the `app` query parameter is set to `foo`, the request should be forwarded to the `foo-app` application.\
+[2] `headers`: if the `x-app` header is set to `bar`, the request should be forwarded to the `bar-app` application.
+
+Now let’s test query parameter matching:
+```console
+$ curl -s http://http.gwapi.${DOMAIN}?app=foo
+Welcome to the foo application!
+```
+
+Next, let's test header matching:
+```console
+$ curl -sH "x-app: bar" http://http.gwapi.${DOMAIN}
+Welcome to the bar application!
+```
+
+You’ve successfully configured Gateway API to match traffic based on path, query parameters, and headers.
+
+## TLS Termination with Gateway API
+In OpenShift, we have the concept of [TLS termination types](https://docs.openshift.com/container-platform/4.13/networking/routes/secured-routes.html)
+to achieve different forms of secured routing. In this example, we will explore how to create an edge-terminated TLS route with Gateway API.
+
+An edge route in OpenShift terminates TLS encryption at the Ingress Controller before forwarding traffic to the
+destination pod:
+```yaml
+apiVersion: v1
+kind: Route
+metadata:
+  name: http-foo
+  namespace: demo-app
+spec:
+  host: edge.${DOMAIN}
+  to:
+    kind: Service
+    name: foo-app
+  port:
+    targetPort: 8080
+tls:
+  termination: edge
+  insecureEdgeTerminationPolicy: Redirect
+  certificate: <CERT>
+  key: <KEY>
+path: /foo
+```
+In Gateway API, we can translate this concept to terminating the TLS encryption at the Gateway before forwarding traffic
+to the upstream server via the HTTPRoute configuration.
```

Comment:
Better to avoid ambiguous words like "upstream", which different implementations use to mean opposite things.
```suggestion
to the destination pod via the HTTPRoute configuration.
```

Reply:
Done